docs/migration_guide.md

```diff
@@ -0,0 +1,31 @@
+---
+title: Migration Guide
+keywords: migration guide
+sidebar: mydoc_sidebar
+permalink: migration_guide.html
+toc: false
+summary: In this page, we will discuss some available tools for migrating your existing dataset into a Hudi managed 
+dataset
+
+
+Hudi maintains metadata such as commit timeline and indexes to manage a dataset. The commit timelines helps to understand the actions happening on a dataset as well as the current state of a dataset. Indexes are used by Hudi to maintain a record key to file id mapping to efficiently locate a record. At the moment, Hudi supports writing only parquet columnar formats.
+To be able to start using Hudi for your existing dataset, you will need to migrate your existing dataset into a Hudi managed dataset. There are a couple of ways to achieve this.
+
+
+## Approaches
+
+
+### Approach 1
+
+Hudi can be used to manage an existing dataset without affecting/altering the historical data already present in the dataset. Hudi has been implemented to be compatible with such a mixed dataset with a caveat that either the complete Hive partition is Hudi managed or not. Thus the lowest granularity at which Hudi manages a dataset is a Hive partition. Start using the datasource API or the WriteClient to write to the dataset and make sure you start writing to a new partition. Note, since the historical partitions are not managed by HUDI, none of the primitives provided by HUDI work on the data in those partitions. More concretely, one cannot perform upserts or incremental pull on such older partitions not managed by the HUDI dataset. 
```

Comment:
"make sure you start writing to a new partition or convert your last N partitions into Hudi instead of entire table"?

Reply:
done